src/gc/impl/manual/gc.d

```diff
@@ -0,0 +1,216 @@
+/**
+ * This module contains a minimal garbage collector implementation according to
+ * published requirements.  This library is mostly intended to serve as an
+ * example, but it is usable in applications which do not rely on a garbage
+ * collector to clean up memory (ie. when dynamic array resizing is not used,
+ * and all memory allocated with 'new' is freed deterministically with
+ * 'delete').
+ *
+ * Please note that block attribute data must be tracked, or at a minimum, the
+ * FINALIZE bit must be tracked for any allocated memory block because calling
+ * rt_finalize on a non-object block can result in an access violation.  In the
+ * allocator below, this tracking is done via a leading uint bitmask.  A real
+ * allocator may do better to store this data separately, similar to the basic
+ * GC.
+ *
+ * Copyright: Copyright Sean Kelly 2005 - 2009.
+ * License:   $(WEB www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
+ * Authors:   Sean Kelly
+ */
+
+/*          Copyright Sean Kelly 2005 - 2009.
+ * Distributed under the Boost Software License, Version 1.0.
+ *    (See accompanying file LICENSE or copy at
+ *          http://www.boost.org/LICENSE_1_0.txt)
+ */
+module gc.impl.manual.gc;
+
+import gc.config;
+import gc.stats;
+import gc.proxy;
+
+import cstdlib = core.stdc.stdlib : calloc, free, malloc, realloc;
+
+static import core.memory;
+private alias BlkAttr = core.memory.GC.BlkAttr;
+private alias BlkInfo = core.memory.GC.BlkInfo;
+
+extern (C) void onOutOfMemoryError(void* pretend_sideffect = null) @trusted pure nothrow @nogc; /* dmd @@@BUG11461@@@ */
+
+import core.stdc.stdio : printf;
+
+struct Range
+{
+    void*  pos;
+    size_t len;
+    TypeInfo ti; // should be tail const, but doesn't exist for references
+}
+
+__gshared GC gcInstance;
+__gshared Proxy  pthis;
+
+struct GC
+{
+
+
+    __gshared void** roots  = null;
+    __gshared size_t nroots = 0;
+
+    __gshared Range* ranges  = null;
+    __gshared size_t nranges = 0;
+
+    void initialize()
+    {
+        if(config.gc != "manual")
```

Comment:
I don't agree with Martin here (but not very strongly). I think it's better to make the decision which GC implementation to initialize in gc_init, because it's more obvious that only one of them is being selected.